HeterogeneousCore/AlpakaCore/src/alpaka/EDMetadataAcquireSentry.cc

```diff
@@ -8,18 +8,29 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     EDMetadataAcquireSentry::EDMetadataAcquireSentry(edm::StreamID streamID,
                                                      edm::WaitingTaskWithArenaHolder holder,
                                                      bool synchronize)
-        : EDMetadataAcquireSentry(detail::chooseDevice(streamID), std::move(holder), synchronize) {}
+        : waitingTaskHolder_(std::move(holder)), synchronize_(synchronize) {
```

Comment:
Could the constructor delegation be still used with
```suggestion
        : EDMetadataAcquireSentry(cms::alpakatools::getQueueCache<Queue>().get(detail::chooseDevice(streamID)), std::move(holder), synchronize) {}
```
?

Reply:
I'm not sure how to do that, because `cms::alpakatools::getQueueCache<Queue>().get(...)` returns a `shared_ptr<Queue>`, not a `Queue&`.